bagel/dictionary_models.py

```diff
@@ -1,6 +1,24 @@
-from typing import Dict, Optional, Union
+from typing import Dict, List, Optional, Union
 
-from pydantic import BaseModel, Extra, Field, conlist
+from pydantic import AfterValidator, BaseModel, ConfigDict, Field, RootModel
+from typing_extensions import Annotated
+
+
+# TODO: Test this function?
+def validate_unique_list(values: List[str]) -> List[str]:
+    """
+    Check that provided list is unique.
+
+    This custom validator is needed because constrained dtypes and their `unique_items` parameter
+    were deprecated in Pydantic v2. This function was adapted from https://github.com/pydantic/pydantic-core/pull/820#issuecomment-1656228704
+    and https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators.
```

Comment:
Good find, how about we use an explicit exception rather than the `assert`?

e.g. like here: https://github.com/pydantic/pydantic-core/pull/820#issuecomment-1670475909

Reply:
Sounds good. For context, the assert syntax is what's used in the Pydantic docs [examples](https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators) - I believe by virtue of living inside an `AfterValidator`, they raise a Pydantic validation error by default.
That said, I think being more explicit here doesn't hurt! 🙂